packages/template-retail-react-app/app/page-designer/README.md

```diff
@@ -59,32 +62,24 @@ const PAGEDESIGNER_TO_COMPONENT = {
     'commerce_layouts.mobileGrid3r2c': MobileGrid3r2c
 }
 
-const PageViewer = ({page}) => (
-    <Box layerStyle={'page'}>
-        <Page page={page} components={PAGEDESIGNER_TO_COMPONENT} />
-    </Box>
-)
-
-PageViewer.getProps = async ({api, params}) => {
-    const {pageId} = params
-    const page = await api.shopperExperience.getPage({
-        parameters: {pageId}
-    })
-
-    if (page.isError) {
-        let ErrorClass = page.type?.endsWith('page-not-found') ? HTTPNotFound : HTTPError
-        throw new ErrorClass(page.detail)
+const PageViewer = () => {
+    const {pageId} = useParams()
+    const {data: page, error}= usePage({parameters: {pageId}})
+
+    if (error) {
+        let ErrorClass = error.response?.status === 404 ? HTTPNotFound : HTTPError
+        throw new ErrorClass(error.response?.statusText)
     }
 
-    return {page}
+    return (
+        <Box layerStyle={'page'}>
+            <Page page={page} components={PAGEDESIGNER_TO_COMPONENT} />
+        </Box>
+    )
 }
 
 PageViewer.displayName = 'PageViewer'
 
-PageViewer.propTypes = {
-    page: pageType.isRequired
-}
-
 export default PageViewer
 ```
 
```

Comment:
Nit: You may want to mention for people unfamiliar with react-router, the route needs to be added before the catch-all route.
https://github.com/SalesforceCommerceCloud/pwa-kit/blob/83f5d54bb1c06cd289e6522b9bb47793509550ab/packages/template-retail-react-app/app/routes.jsx#L100-L103

Reply:
🤔 I feel like at this point in time the developer should be aware as to what a router is and how it's used by going through the doc's on developer.salesforce.com. I'm hesitant to add router specific documentation outside of the context of the router incase it ever changes we might have to go back and change all the other doc's instead of only changing one place.